libsolidity/formal/CHC.cpp

```diff
@@ -61,11 +61,59 @@ bool CHC::visit(ContractDefinition const& _contract)
 	if (!shouldVisit(_contract))
 		return false;
 
+	m_contract = &_contract;
+
 	reset();
 
 	if (!SMTEncoder::visit(_contract))
 		return false;
 
+	m_stateVariables = stateVariables(_contract);
+
+	for (auto const& var: m_stateVariables)
+		// SMT solvers do not support function types as arguments.
+		if (var->type()->category() == Type::Category::Function)
+			m_stateSorts.push_back(make_shared<smt::Sort>(smt::Kind::Int));
+		else
+			m_stateSorts.push_back(smt::smtSort(*var->type()));
+
+	string interfaceName = "interface_" + _contract.name() + "_" + to_string(_contract.id());
+	m_interfacePredicate = createBlock(interfaceSort(),	interfaceName);
+
+	// TODO create static instances for Bool/Int sorts in SolverInterface.
+	auto boolSort = make_shared<smt::Sort>(smt::Kind::Bool);
+	auto errorFunctionSort = make_shared<smt::FunctionSort>(
+		vector<smt::SortPointer>(),
+		boolSort
+	);
+	m_errorPredicate = createBlock(errorFunctionSort, "error");
+
+	// If the contract has a constructor it is handled as a function.
+	// Otherwise we zero-initialize all state vars.
```

Comment:
Don't we need to do that in both cases?

Reply:
Not sure, BMC wouldn't really make use of initialized state vars while checking functions